app/docker/views/host/host-view-controller.js

```diff
@@ -24,6 +24,11 @@ angular.module('portainer.docker').controller('HostViewController', [
           ctrl.hostDetails = buildHostDetails(data.info);
 
           if (ctrl.state.isAgent) {
+            var agentApiVersion = applicationState.endpoint.agentApiVersion;
```

Comment:
What about using a conditional request in the `$q.all()` instead ? I think it's a lot clearer.

Example: https://github.com/portainer/portainer/blob/develop/app/docker/views/dashboard/dashboardController.js#L18